Atomic/rpmwriter.py

```diff
@@ -0,0 +1,369 @@
+    SIGNATURE_VERSION = [0x01]
+    SIGNATURE_RESERVED = [0x0] * 4
+
+    RPMSIGTAG_PAYLOADSIZE = 1007
+
+    RPMTAG_NAME = 1000
+    RPMTAG_VERSION = 1001
+    RPMTAG_RELEASE = 1002
+    RPMTAG_SUMMARY = 1004
+    RPMTAG_DESCRIPTION = 1005
+    RPMTAG_LICENSE_ = 1014
+    RPMTAG_GROUP = 1016
+    RPMTAG_CHANGELOG = 1017
+    RPMTAG_URL = 1020
+    RPMTAG_OS = 1021
+    RPMTAG_ARCH = 1022
+    RPMTAG_FILESIZES = 1028
+    RPMTAG_DIRINDEXES = 1116
+    RPMTAG_BASENAMES = 1117
+    RPMTAG_DIRNAMES = 1118
+    RPMTAG_FILEMODES = 1030
+    RPMTAG_FILERDEVS = 1033
+    RPMTAG_FILEMTIMES = 1034
+    RPMTAG_FILEDIGESTS = 1035
+    RPMTAG_FILELINKTOS = 1036
+    RPMTAG_FILEFLAGS = 1037
+    RPMTAG_FILEUSERNAME = 1039
+    RPMTAG_FILEGROUPNAME = 1040
+
+    RPMTAG_PROVIDENAME		= 1047
+    RPMTAG_REQUIRENAME		= 1049
+    RPMTAG_REQUIREVERSION	= 1050
+    RPMTAG_CONFLICTFLAGS	= 1053
+    RPMTAG_CONFLICTNAME		= 1054
+    RPMTAG_CONFLICTVERSION	= 1055
+
+    RPMTAG_OBSOLETENAME		= 1090
+    RPMTAG_FILEINODES		= 1096
+    RPMTAG_FILELANGS		= 1097
+
+    RPMTAG_PAYLOADFORMAT	= 1124
+    RPMTAG_PAYLOADCOMPRESSOR	= 1125
+
+    RPMTAG_FILEDIGESTALGO = 5011
+
+    RPM_SPEC_FILEMODE	= (1 << 8)
+    RPM_SPEC_DIRMODE	= (1 << 9)
+
+    RPMFILE_CONFIG	= (1 <<  0)
+
+    RPMTAG_PAYLOADDIGEST	= 5092
+    RPMTAG_PAYLOADDIGESTALGO	= 5093
+
+    PGPHASHALGO_SHA1 =  2
+
+    def get_sha1(self, path):
+        m = hashlib.sha1()
+        try:
+            with open(path, 'r') as f:
+                while True:
+                    data = os.read(f.fileno(), 4096)
+                    if len(data) == 0:
+                        break
+                    m.update(data)
+            return m.hexdigest()
+        except IOError:
+            return ""
+
+    def add_require(self, name, version):
+        self.require.append([name, version])
+
+    def add_provide(self, name):
+        self.provide.append(name)
+
+    def add_obsolete(self, name):
+        self.obsolete.append(name)
+
+    def add_conflict(self, name, version):
+        self.conflict.append([name, version])
+
+    def __init__(self, out, root, name, version, release, summary='', description='', license_='gpl2', changelog='', url='', group='', stderr=sys.stderr, whitelist=None):
+        self.out = out
+        self.name = name
+        self.version = version
+        self.release = release
+        self.headers = []
+        self.written = 0
+        self.root = root
+        self.all_files = []
+        self.require = []
+        self.provide = []
+        self.obsolete = []
+        self.conflict = []
+        self.summary = summary
+        self.description = description
+        self.license_ = license_
+        self.changelog = changelog
+        self.url = url
+        self.group = group
+        self.stderr = stderr
+        self.whitelist = set(whitelist) if whitelist is not None else None
+
+    def add_header(self, tag, typ, count, value, pad=1):
+        try:
+            # this fails on python3
```

Comment:
Ah, I read it to mean you were having unexpected issues here.  Perhaps change the comment to something like:

"The next line is expected to always fail under python3, in that case just continue."

or something similar.  Another nit, not a hold the merge comment.